### ISSUE TITLE
2.0.0 Release

### ISSUE THREAD

**Issue by metaskills** [rails-lambda/crypteia#48]
No need for a PR, this work can land in main. Making this so we can communicate around this task.

**Comment by metaskills** [comment]
@jeremiahlukus Can you take a look at the two latest commits on main? See how I removed the latest temporarily so we can build a 2.0.0.pre release without pushing latest? Thoughts? Also, can you help with the CHANGELOG?

**Comment by jeremiahlukus** [comment]
Yes, and yes ill get it sometime today

**Comment by jeremiahlukus** [comment]
Going to send another PR to update the cargo packages as well

**Comment by jeremiahlukus** [comment]
https://github.com/rails-lambda/crypteia/actions/runs/9488709142

Released

will start testing tomorrow working on something else right now.

**Comment by jeremiahlukus** [comment]
Have some error will have to do another release 


```
0.078 /bin/sh: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /opt/lib/libcrypteia.so)
0.078 /bin/sh: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/lib/libcrypteia.so)
```

**Comment by jeremiahlukus** [comment]
https://github.com/rust-lang/rust/issues/57497

According to this we are kind of stuck building in this older docker container. 
Ill file another issue to update this image but not sure what we can do about it.

**Comment by jeremiahlukus** [comment]
Same error after reverting the base img 


```
0.101 /bin/sh: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /opt/lib/libcrypteia.so)
0.101 /bin/sh: /lib/aarch64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /opt/lib/libcrypteia.so)
```

**Comment by jeremiahlukus** [comment]
Ok checking my ruby img from `ruby:3.2.2-bullseye` to `ruby:3.2.2-bookworm` fixed it I guess bookworm has GLIBC installed. 

 @metaskills  I do not really know how to fix this on the  crypteia side 
 
 
 Now that I am able to build the docker image though I can going to test.

**Comment by jeremiahlukus** [comment]
Ok just tested its not working as expected. Going to do releases from my branch. I thought returning an error would do the trick but its a bit more complicated

**Comment by jeremiahlukus** [comment]
O i see the path issue 

```
            (
                "X_CRYPTEIA_SSM".to_string(),
                "x-crypteia-ssm-path:/crypteia/v5/myapp/envs".to_string(),
            ),
            // ("DB_URL".to_string(), "x-crypteia".to_string()),
            // ("NR_KEY".to_string(), "x-crypteia".to_string()),
            
```
https://github.com/rails-lambda/crypteia/blob/main/src/ssm.rs#L180


I misunderstood from the readme 

```
    X_CRYPTEIA_SSM: x-crypteia-ssm-path:/myapp/envs
    DB_URL: x-crypteia
    NR_KEY: x-crypteia
```

It seems like 
```
    DB_URL: x-crypteia
    NR_KEY: x-crypteia
```

is not needed its just for show.

**Comment by jeremiahlukus** [comment]
I agree with latest , I will delete all my leftovers after 👍. Now that I can test and release on my branch I don’t need to spam commits to master.

**Comment by jeremiahlukus** [comment]
tl:dr: I reverted to v1 code I do not agree with the v2 changes after testing them.



I reverted back to v1 code for main and ssm. 
Right now v2 will error out the lambda if an ssm isnt found. I could process exit and try to reboot the lambda but then it would just spin up another and be in a loop. 

Then i thought i could hook into the deploy since i found  after setting LD_PRELOAD crypteia just runs in the background immediately. So i thought if i set 

```
    if std::env::var("AWS_LAMBDA_FUNCTION_NAME").is_err() {
        log::cloudwatch_metric("main", "running_outside_lambda", false, None);
        println!("Running outside of AWS Lambda environment.");
    }
 ```
 
 This would get ran then i could run some prechecks. This would be way too many assumptions on how things outside this app are. 
 
 
 
 For me I would rather have the v1 code, where the app doesn't break if it cant find a param. 
 Then in my deploy script have a precheck for params (which is what i normally do)
 
 ```
 echo "== Parsing template.yaml for x-crypteia-ssm variables =="
SSM_VARS=$(awk -F'"' '/x-crypteia-ssm:\/myapp\/\$\{RailsEnv\}\// {gsub(/.*x-crypteia-ssm:\/myapp\/\$\{RailsEnv\}\//, "", $2); print $2}' template.yaml)

for var in $SSM_VARS; do
  SSM_PATH="/myapp/${RAILS_ENV}/${var}"
  echo "Checking SSM parameter: $SSM_PATH"
  aws ssm get-parameter --name "$SSM_PATH" --region "$AWS_REGION" > /dev/null
  if [ $? -ne 0 ]; then
    echo "SSM parameter $SSM_PATH is not present."
    exit 1
  fi
done
```

I cant think of a graceful way for crypteia to handle ssm vars that are not present after exploring several options I think the best thing is a script to verify params are there and trust crypteia to do its thing.  

On a positive note the build process is fixed and I can now say I know how this app works and I am able to help add features and debug issues haha. 
First time in a rust application so was a little lost, especially since this isn't a normal app.

**Comment by jeremiahlukus** [comment]
https://github.com/rails-lambda/crypteia/commit/d343e23e520121f53d8366ef7384972c15d9a474

Going to release this as 2.0.0

**Comment by metaskills** [comment]
> is not needed its just for show.

Yup, this was a security measure to be explicit about what in the path could be applied to the env.

> tl:dr: I reverted to v1 code I do not agree with the v2 changes after testing them.

👀 

> but then it would just spin up another and be in a loop

Yup, I kind of assumed that would be the case but have not put in the leg work to think thru how this works in practice and the things folks need to do to ensure the service is not doing anything crazy. Like making more SSM calls and going over quota. Stuff like that.

> I cant think of a graceful way for crypteia to handle ssm vars that are not present after exploring several options I think the best thing is a script to verify params are there and trust crypteia to do its thing.

This aligns with my thinking as well. Honestly, I appreciate you exploring it too. cc @drinks 

> On a positive note the build process is fixed and I can now say I know how this app works and I am able to help add features and debug issues haha. First time in a rust application so was a little lost, especially since this isn't a normal app.

Dude, you HAVE DONE AN AMAZING JOB!

I cant thank you enough. Honestly. It feels so good to have the pipeline working again. Remember, everything is progress. Even learning. Well done.
